modules/channel-telegram/src/renderers/carousel.ts

```diff
@@ -0,0 +1,55 @@
+import * as sdk from 'botpress/sdk'
+import { ChannelRenderer } from 'common/channel'
+import { formatUrl } from 'common/url'
+import path from 'path'
+import { Markup } from 'telegraf'
+import Extra from 'telegraf/extra'
+import { TelegramContext } from '../backend/typings'
+
+export class TelegramCarouselRenderer implements ChannelRenderer<TelegramContext> {
+  get channel(): string {
+    return 'telegram'
+  }
+
+  get priority(): number {
+    return 0
+  }
+
+  get id() {
+    return TelegramCarouselRenderer.name
+  }
+
+  handles(context: TelegramContext): boolean {
+    return context.payload.items?.length
+  }
+
+  render(context: TelegramContext) {
+    const { messages } = context
+    const payload = context.payload as sdk.CarouselContent
+
+    for (const card of payload.items) {
+      if (card.image) {
+        messages.push({ action: 'upload_photo' })
+        messages.push({ photo: { url: formatUrl(context.botUrl, card.image), filename: path.basename(card.image) } })
+      }
+
+      const buttons = []
+      for (const action of card.actions) {
+        if (action.action === 'Open URL') {
```

Comment:
I know it's really picky, but can you change the typing of card.actions so that the action.action is strongly typed. Since there is a finite number of values this property can take, I think it would be better if you used something like: `type Action = 'Open URL' | 'Say something' | 'Postback'` instead of `string`. I think it would be less error-prone. What do you think?